[[provide]] node in build plan

jabrown85 opened the pull request:
Ensure that the buildpack defines what it provides so that other buildpacks that depend on this buildpack can specify that requirement. The reason `node` was chosen was to align with the name that the riff node buildpack uses to require `node`.

https://github.com/buildpacks/spec/blob/master/buildpack.md#process
https://github.com/projectriff/node-function-buildpack/blob/07eacdbe9fe49b8ce8f726fc617a5bc895b9a7b4/node/buildpack.go#L72

danielleadams commented on bin/detect:
```diff
@@ -5,10 +5,16 @@ set -eo pipefail
+  cat << EOF > "$build_plan"
+  [[provides]]
+  name = "node"
+EOF
```
Can we do the same thing here as requested in the npm buildpack: https://github.com/heroku/nodejs-npm-buildpack/pull/18#discussion_r372547386

Then we can add a unit test for it.

danielleadams commented:
@jabrown85 Do you mind updating the Changelog for this too?

jabrown85 commented:
@danielleadams I bumped the version, updated changelog, and added a test. I also tested that this buildpack works with the builders as-is. Let me know if you want me to test anything else. I'll follow up on the npm-modules one next.

danielleadams commented:
Looks good - thanks for the changes! I'm going to go ahead and merge this, but I'm going to add one more change to this release before I tag it.
